src/envConfig.ts

```diff
@@ -2,54 +2,60 @@ import fs from 'fs';
 import path from 'path';
 import type { Plugin } from 'vite';
 
-/**
- * This plugin serves the `/assets/env-config.js` file with proper environment variables
- */
-
-export function renderTemplateWithEnvVars(content: string): string {
-  let result = content;
-  const regex = /\$\{([a-zA-Z_]+[a-zA-Z0-9_]*?)\}/g;
-
-  let matched: RegExpExecArray | null;
-  while ((matched = regex.exec(content))) {
-    result = result.replace(`\${${matched[1]}}`, process.env[matched[1]] || '');
-  }
+export function createEnvConfigContent(variables: string[], template: boolean): string {
+  let templateContent = '';
+  templateContent += '(function(window){window.env=window.env||{};';
+  variables.forEach((variable) => {
+    if (template) {
+      templateContent += `window.env['${variable}']='\${${variable}}';`;
+    } else {
+      templateContent += `window.env['${variable}']='${process.env[variable] || ''}';`;
```

Comment:
```suggestion
      templateContent += `window.env['${variable}']=${process.env[variable] ? `'${process.env[variable]}'` : 'undefined'};`;
```

Reply:
@anbraten What do you want to achieve here?

The current implementation matches the behavior of `envsubst` to just insert an empty string when the variable is not set. I think matching behavior in development and production are quite important.